open_schools_platform/organization_management/circles/tests/views/test_circle_invite_teacher.py

```diff
@@ -0,0 +1,122 @@
+from open_schools_platform.query_management.queries.selectors import get_query
+from open_schools_platform.user_management.users.tests.utils import create_logged_in_user, create_test_user
+
+
+class InviteTeacherTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.invite_teacher_url = \
+            lambda pk: reverse("api:organization-management:circles:invite-teacher", args=[pk])
+
+    @pytest.mark.skipif(env.ENVIRON.get("EMAIL_ID") is None or env.ENVIRON.get("EMAIL_PRIVATE_API_KEY") is None or
+                        env.ENVIRON.get("EMAIL_ID") == "" or env.ENVIRON.get("EMAIL_PRIVATE_API_KEY") == "",
+                        reason="EMAIL_ID or EMAIL_PRIVATE_API_KEY is not set")
+    def test_invite_teacher_query_successfully_formed_1(self):
+        user = create_logged_in_user(self)
+        circle = create_test_circle_with_user_in_org(user)
+        data = create_data_circle_invite_teacher("TestTeacher", "+79998881177")
+
+        response = self.client.post(self.invite_teacher_url(str(circle.id)), data, format="json")
+        self.assertEqual(201, response.status_code)
+        teacher_profile = get_teacher_profile(filters={"phone": data["phone"]})
+        self.assertTrue(teacher_profile)
+
+    def test_invite_teacher_query_successfully_formed_2(self):
+        user = create_logged_in_user(self)
+        circle = create_test_circle_with_user_in_org(user)
+
+        teacher_profile = create_test_user("+79998786644").teacher_profile
+
+        data = create_data_circle_invite_teacher("TestTeacher", str(teacher_profile.user.phone))
+
+        response = self.client.post(self.invite_teacher_url(str(circle.id)), data, format="json")
+        self.assertEqual(201, response.status_code)
+
+
+class CircleInviteTeacherQueryTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.query_status_change_url = reverse("api:query-management:queries:change-query-status")
+
+    def test_teacher_can_change_status_on_accepted(self):
+        circle = create_test_circle()
+        user = create_logged_in_user(self)
+        teacher_profile = user.teacher_profile
+        teacher = create_teacher(teacher_profile=teacher_profile, circle=circle, name='test_teacher')
+        query = create_test_query_circle_invite_teacher(circle, teacher, teacher_profile)
+        data = {
+            "id": query.id,
+            "status": Query.Status.ACCEPTED
+        }
+        response = self.client.patch(self.query_status_change_url, data)
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(Query.Status.ACCEPTED, get_query(filters={"id": str(query.id)}).status)
+
+    def test_teacher_can_change_status_on_declined(self):
+        circle = create_test_circle()
+        user = create_logged_in_user(self)
+        teacher_profile = user.teacher_profile
+        teacher = create_teacher(teacher_profile=teacher_profile, circle=circle, name='test_teacher')
```

Comment:
^^

Reply:
You are creating such teacher in every test, so I won't point it out down below, but, please, check them as well.